src/seedu/addressbook/data/person/Name.java

```diff
@@ -66,6 +67,8 @@ public int hashCode() {
      * Two names are considered similar if ...
      */
      public boolean isSimilar(Name other) {
-    	 return other.equals(this) || other.toString().toLowerCase().equals(this.toString().toLowerCase());
+    	 return other.equals(this)
+    			 || other.toString().toLowerCase().equals(this.toString().toLowerCase())
+    			 || new HashSet<>(Arrays.asList(other.toString().split(" "))).equals(new HashSet<>(Arrays.asList(this.toString().split(" "))));
```

Comment:
Avoid lines longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.